fenix/app/src/main/java/org/mozilla/fenix/onboarding/AdjustActivatedUserWorker.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding
+
+import android.content.Context
+import androidx.work.ExistingWorkPolicy
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.google.android.gms.common.util.VisibleForTesting
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Worker that tracks conditions for "Activated" event for new users.
+ */
+class AdjustActivatedUserWorker(
```

Comment:
This `Worker` looks to be doing pretty similar things to what is happening in the `MetricsMiddleware`. I don't necessarily have a strong opinion on using one over the other, but I think we should avoid splitting how we handle events like these into multiple types of solutions. I think it would probably be easier to remove this and use the `MetricsMiddleware`, but I am also open to having the events in the Middleware moved into the Worker instead. Since `WorkManager`'s work queue persists across things like process death, this would improve our data integrity, though I am not sure whether we know that that is actually needed or not.

If we do want to move the other events here, I think it would be fine to do that as a follow-up.

Reply:
I agree, I removed this worker and moved some of the logic to `MetricsMiddleware` for the "usage" counter.